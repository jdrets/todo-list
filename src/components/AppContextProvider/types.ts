import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
