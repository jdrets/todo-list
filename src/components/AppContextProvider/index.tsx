import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
