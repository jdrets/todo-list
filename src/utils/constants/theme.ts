const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
