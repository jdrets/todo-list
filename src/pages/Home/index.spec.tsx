import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'


import HomePage from '.'
import theme from '../../utils/constants/theme'
import AppContextProvider from '../../components/AppContextProvider'

const setup = () => (
  render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <HomePage />
      </AppContextProvider>
    </ThemeProvider>
  )
)

describe('<HomePage />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
