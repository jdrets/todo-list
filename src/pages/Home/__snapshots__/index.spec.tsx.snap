// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match with snapshot 1`] = `
<div>
  <section
    class="sc-bdnylx jACAPs"
  >
    <section
      class="sc-iqAbSa cfJmRf"
    >
      <section />
      <button
        class="sc-crzoUp fEURFJ"
      >
        <span
          class="sc-eCApGN dQCaZs"
          color="#000000"
          height="22px"
          rotate="0"
          width="22px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M472 168H40a24 24 0 010-48h432a24 24 0 010 48zM392 280H120a24 24 0 010-48h272a24 24 0 010 48zM296 392h-80a24 24 0 010-48h80a24 24 0 010 48z"
            />
          </svg>
        </span>
        <span
          class="sc-dIsAE hQUSxL"
        >
          Filtros
        </span>
      </button>
    </section>
    <article
      class="sc-gtssRu jEMKxY"
    >
      <div
        class="sc-jSFkmK lgjONk"
      >
        <div
          class="sc-gKAblj jSBQkU"
        >
          <div
            class="sc-iCoHVE eIXjhW"
          >
            <span
              class="sc-eCApGN dQCaZs"
              color="#000000"
              height="22px"
              rotate="0"
              width="22px"
            >
              <svg
                class=""
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                
                <path
                  d="M307.72 302.27a8 8 0 01-3.72-6.75V80a48 48 0 00-48-48h0a48 48 0 00-48 48v215.52a8 8 0 01-3.71 6.74 97.51 97.51 0 00-44.19 86.07A96 96 0 00352 384a97.49 97.49 0 00-44.28-81.73zM256 112v272"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-miterlimit="10"
                  stroke-width="32"
                />
                <circle
                  cx="256"
                  cy="384"
                  r="48"
                />
              </svg>
            </span>
          </div>
          <div
            class="sc-fujyUd hTzWJH"
          >
            Priority: 
            <span
              class="sc-pNWxx hxYwIh"
            >
              High
            </span>
          </div>
        </div>
        <div
          class="sc-gKAblj jSBQkU"
        >
          <div
            class="sc-iCoHVE eIXjhW"
          >
            <span
              class="sc-eCApGN dQCaZs"
              color="#000000"
              height="22px"
              rotate="0"
              width="22px"
            >
              <svg
                class=""
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                
                <path
                  d="M394.23 197.56a300.43 300.43 0 00-53.37-90C301.2 61.65 249.05 32 208 32a16 16 0 00-15.48 20c13.87 53-14.88 97.07-45.31 143.72C122 234.36 96 274.27 96 320c0 88.22 71.78 160 160 160s160-71.78 160-160c0-43.3-7.32-84.49-21.77-122.44zm-105.9 221.13C278 429.69 265.05 432 256 432s-22-2.31-32.33-13.31S208 390.24 208 368c0-25.14 8.82-44.28 17.34-62.78 4.95-10.74 10-21.67 13-33.37a8 8 0 0112.49-4.51A126.48 126.48 0 01275 292c18.17 24 29 52.42 29 76 0 22.24-5.42 39.77-15.67 50.69z"
                />
              </svg>
            </span>
          </div>
          <div
            class="sc-fujyUd hTzWJH"
          >
            Status: 
            <span
              class="sc-jrsJCI hinfDw"
            >
              Doing
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-dlnjPT KSUMJ"
      >
        Deploy App
      </div>
      <div
        class="sc-hKFyIo kJaGaR"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse molestie mollis mauris vitae sagittis. Suspendisse dictum turpis lacus, eu sodales nisl commodo et. Donec ultricies dui quis metus facilisis, porta eleifend ante egestas. Cras sed convallis odio. Nam fermentum sit amet velit at consectetur. Cras vitae bibendum justo, eu sagittis nunc. Sed at quam arcu. Sed consectetur at mi a fringilla.
      </div>
    </article>
  </section>
</div>
`;
